File: src/main/java/de/flapdoodle/generics/uniontypes/j/components/TextField.java
package de.flapdoodle.generics.uniontypes.j.components;

import de.flapdoodle.generics.uniontypes.j.framework.Component;
import de.flapdoodle.generics.uniontypes.j.framework.HasValidation;
import de.flapdoodle.generics.uniontypes.j.framework.HasValue;

public class TextField extends Component implements HasValue<String>, HasValidation {
  private String value = null;
  private boolean invalid= false;

  @Override
  public String getValue() {
    return value;
  }

  @Override
  public void setValue(String value) {
    this.value = value;
  }

  @Override
  public void setInvalid(Boolean flag) {
    this.invalid = flag;
  }
}

File: src/main/kotlin/de/flapdoodle/generics/uniontypes/k/components/TextField.kt
package de.flapdoodle.generics.uniontypes.k.components

import de.flapdoodle.generics.uniontypes.k.framework.Component
import de.flapdoodle.generics.uniontypes.k.framework.HasValidation
import de.flapdoodle.generics.uniontypes.k.framework.HasValue

class TextField : Component(), HasValue<String>, HasValidation {
  private var value: String? = null
  private var invalid: Boolean = false

  override fun getValue(): String? = value

  override fun setValue(value: String?) {
    this.value = value
  }

  override fun setInvalid(flag: Boolean) {
    this.invalid = flag
  }
}